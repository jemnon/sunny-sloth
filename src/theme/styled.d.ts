import 'styled-components';
import {
  Breakpoints,
  Colors,
  ColorMode,
  FocusColors,
  Fonts,
  FontSizes,
  Header,
  HoverColors,
  MaxWidths,
  Spacing,
  ZIndex,
} from '../types/theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    breakpoints: Breakpoints;
    colors: Colors;
    focusColors: FocusColors;
    fonts: Fonts;
    fontSizes: FontSizes;
    header: Header;
    hoverColors: HoverColors;
    maxWidths: MaxWidths;
    spacing: Spacing;
    transition: string;
    zIndex: ZIndex;
  }
}
