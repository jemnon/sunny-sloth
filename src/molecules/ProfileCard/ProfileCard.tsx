import React, { FC, ReactNode } from 'react';
import styled from 'styled-components';
import Avatar from '../../atoms/Avatar';
import Text from '../../atoms/Text';
import CalloutLink from '../CalloutLink';

interface ProfileCardProps {
  descriptionHtml?: string;
  name: string;
  image: string;
  onClick?: () => void;
}

const ProfileCardContainer = styled.section`
  display: flex;
  align-items: center;
  flex-direction: column;

  padding-top: ${({ theme }): string => theme.spacing.md4};
  padding-bottom: ${({ theme }): string => theme.spacing.md4};
  padding-left: ${({ theme }): string => theme.spacing.sm4};
  padding-right: ${({ theme }): string => theme.spacing.sm4};

  background-color: ${({ theme }): string => theme.colors.nearWhite};
`;

const ProfileCardName = styled.h5`
  margin-bottom: ${({ theme }): string => theme.spacing.sm4};
`;

const ProfileCardDivider = styled.div`
  width: 45px;
  height: 4px;

  background-color: ${({ theme }): string => theme.colors.orange};

  margin-bottom: ${({ theme }): string => theme.spacing.sm4};
`;

const ProfileCardDescription = styled.div`
  font-size: ${({ theme: { fontSizes } }): string => fontSizes.f1};
  p {
    font-size: ${({ theme: { fontSizes } }): string => fontSizes.f1};
    text-align: center;
  }

  margin-bottom: ${({ theme }): string => theme.spacing.sm4};
`;

const ProfileFooter = styled.footer`
  display: flex;
  justify-content: flex-end;

  width: 100%;
`;

const ProfileCard: FC<ProfileCardProps> = ({
  descriptionHtml,
  name,
  image,
  onClick,
}) => (
  <ProfileCardContainer>
    <Avatar alt="jerildine" src={image} />
    <ProfileCardName as={Text} fontSize="f4" fontWeight="bold">
      {name}
    </ProfileCardName>
    <ProfileCardDivider />
    <ProfileCardDescription
      dangerouslySetInnerHTML={{
        __html: descriptionHtml ?? '',
      }}
    />

    <ProfileFooter>
      <CalloutLink onClick={onClick}>Read More</CalloutLink>
    </ProfileFooter>
  </ProfileCardContainer>
);

export default ProfileCard;
