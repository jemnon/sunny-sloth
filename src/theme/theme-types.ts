export interface Breakpoints {
  sm: string;
  md: string;
  lg: string;
  xlg: string;
}

export interface Fonts {
  lato: string;
  latoBold: string;
  latoItalic: string;
  noto: string;
  notoBold: string;
}

export interface Colors {
  cream: string;
  darkTeal: string;
  teal: string;
  lightBrown: string;
  nearBlack: string;
  nearWhite: string;
  darkOrange: string;
  orange: string;
  white: string;
  red: string;
  green: string;
}

export type ColorMode = 'dark' | 'light';

export interface MaxWidths {
  sm: string;
  md: string;
  lg: string;
}

export interface Spacing {
  sm1: string;
  sm2: string;
  sm3: string;
  sm4: string;
  md1: string;
  md2: string;
  md3: string;
  md4: string;
  lg1: string;
  lg2: string;
  lg3: string;
  lg4: string;
}

export interface ZIndex {
  z0: string;
  z1: string;
  z2: string;
  z3: string;
  z4: string;
  z5: string;
  z6: string;
  z999: string;
  z9999: string;
}
