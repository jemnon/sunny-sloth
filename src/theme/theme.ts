const theme = {
  breakpoints: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
    xlg: '1440px',
  },
  fonts: {
    latoBold:
      `"Lato-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato-Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    notoBold:
      `"Noto Serif TC-Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1280px',
  },
  spacing: {
    sm1: '0.25rem' /* 4px */,
    sm2: '0.5rem' /*  8px */,
    sm3: '0.75rem' /* 12px */,
    sm4: '1rem' /* 16px */,
    md1: '1.25rem' /* 20px */,
    md2: '1.5rem' /* 24px */,
    md3: '1.75rem' /* 28px */,
    md4: '2rem' /* 32px */,
    lg1: '2.25rem' /* 36px */,
    lg2: '2.5rem' /* 40px */,
    lg3: '2.75rem' /* 44px */,
    lg4: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
