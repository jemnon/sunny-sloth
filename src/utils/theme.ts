import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  breakpoints: {
    /* TODO: remove once conversion to styled-breakpoints is done */
    desktop: 'screen and (min-width: 960px)',
    tablet: 'screen and (min-width: 480px) and (max-width: 959px)',
    mobile: 'screen and (max-width: 479px)',
    /* ^^ */
    sm: '480px',
    md: '960px',
    lg: '1440px',
    xlg: '1920px',
  },
  fonts: {
    latoBold:
      `"Lato Bold", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    lato:
      `"Lato", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    latoItalic:
      `"Lato Italic", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
    noto:
      `"Noto Serif TC", -apple-system, BlinkMacSystemFont, 'avenir next', avenir, 'Helvetica Neue', ` +
      `helvetica, ubuntu, roboto, noto, 'Segoe UI', Arial, sans-serif`,
  },
  colors: {
    cream: '#f4ede6',
    darkTeal: '#162936',
    teal: '#244c53',
    lightBrown: '#d5a188',
    nearBlack: '#111',
    nearWhite: '#F9F9F9',
    darkOrange: '#BB5B34',
    orange: '#cf7651',
    white: '#ffffff',
    red: '#ff4136',
    green: '#19a974',
  },
  maxWidths: {
    sm: '480px',
    md: '960px',
    lg: '1440px',
  },
  spacing: {
    s1: '0.25rem' /* 4px */,
    s2: '0.5rem' /*  8px */,
    s3: '0.75rem' /* 12px */,
    s4: '1rem' /* 16px */,
    s5: '1.25rem' /* 20px */,
    s6: '1.5rem' /* 24px */,
    s7: '1.75rem' /* 28px */,
    s8: '2rem' /* 32px */,
    s9: '2.25rem' /* 36px */,
    s10: '2.5rem' /* 40px */,
    s11: '2.75rem' /* 44px */,
    s12: '3rem' /* 48px */,
  },
  transition: 'all 0.47s ease',
  zIndex: {
    z0: '0',
    z1: '1',
    z2: '2',
    z3: '3',
    z4: '4',
    z5: '5',
    z6: '6',
    z999: '999',
    z9999: '9999',
  },
};

export default theme;
