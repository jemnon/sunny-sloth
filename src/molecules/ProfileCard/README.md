# ProfileCard

### Description
Component description

```js
import ProfileCard from '@fenderdigital/ui/FILL IN TYPEProfileCard';
```

### Props
* `propName` - property name and description 
