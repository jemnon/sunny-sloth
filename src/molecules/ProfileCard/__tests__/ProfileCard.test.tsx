import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import ProfileCard from '../ProfileCard';
import theme from '../../../theme';

describe('<ProfileCard />', (): void => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <ProfileCard
        description="I get down on making all sorts of eats"
        image="https://images.ctfassets.net/lz7g6u6kccw7/2PXqIKP4gcdBKWQ4sUuqaa/c4ea960704b036c285cd6061400c447e/IMG_6D21ECE4038E-1.jpeg?h=250"
      />
    </ThemeProvider>,
  );
  it('exists', (): void => {
    expect(ProfileCard).toBeDefined();
  });
  it('renders defaults correctly', (): void => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
