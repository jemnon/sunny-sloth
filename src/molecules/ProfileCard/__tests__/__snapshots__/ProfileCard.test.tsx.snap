// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCard /> renders defaults correctly 1`] = `
<section
  className="sc-fKVsgm gHITeq"
>
  <img
    alt="jerildine"
    className="sc-bdvvaa dOfXvM"
    src="https://images.ctfassets.net/lz7g6u6kccw7/2PXqIKP4gcdBKWQ4sUuqaa/c4ea960704b036c285cd6061400c447e/IMG_6D21ECE4038E-1.jpeg?h=250"
  />
  <div
    className="sc-gsDJrp gnvuur sc-bBHwJV bRtXXM"
    fontSize="f4"
    fontWeight="bold"
  >
    jerildine
  </div>
  <div
    className="sc-iwjezw CtbIG"
  />
  <div
    className="sc-cxpRKc bjBnP"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
  <footer
    className="sc-llYToB HdhJl"
  >
    <button
      className="sc-dlVyqM bxzSft"
    >
      <span
        className="sc-kfPtPH iLVomG"
      >
        Read More
      </span>
      <svg
        className="sc-egiSv fpsiTh"
        fill="#A2421B"
        fontSize="10px"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "marginTop": "1px",
          }
        }
        viewBox="0 0 9.9 16.97"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.41 0L0 1.41l7.07 7.07L0 15.56l1.41 1.41L8.48 9.9 9.9 8.48 1.41 0z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </footer>
</section>
`;
